internal/staging/stdaddr/addressv0.go

```diff
@@ -648,13 +649,33 @@ func (addr *AddressPubKeyHashEcdsaSecp256k1V0) VotingRightsScript() (uint16, []b
 	return 0, script[:]
 }
 
+// calcRewardCommitScriptLimits calculates the encoded limits to impose on fees
+// applied to votes and revocations via the reward commitment script of a ticket
+// purchase.
+func calcRewardCommitScriptLimits(voteFeeLimit, revocationFeeLimit int64) uint16 {
+	// The limits are defined in terms of the closest base 2 exponent and
+	// a bit that must be set to specify the limit is to be applied.  The
+	// vote fee exponent is in the bottom 8 bits, while the revocation fee
+	// exponent is in the upper 8 bits.
+	limits := uint16(0)
+	if voteFeeLimit != 0 {
+		exp := uint16(math.Ceil(math.Log2(float64(voteFeeLimit))))
+		limits |= (exp | 0x40)
+	}
+	if revocationFeeLimit != 0 {
+		exp := uint16(math.Ceil(math.Log2(float64(revocationFeeLimit))))
+		limits |= ((exp | 0x40) << 8)
+	}
+	return limits
+}
+
 // RewardCommitmentScript returns the script version associated with the address
 // along with a script that commits the original funds locked to purchase a
 // ticket plus the reward to the address along with limits to impose on any
 // fees.
 //
 // This is part of the StakeAddress interface implementation.
-func (addr *AddressPubKeyHashEcdsaSecp256k1V0) RewardCommitmentScript(amount int64, limits uint16) (uint16, []byte) {
+func (addr *AddressPubKeyHashEcdsaSecp256k1V0) RewardCommitmentScript(amount, voteFeeLimit, revocationFeeLimit int64) (uint16, []byte) {
```

Comment:
Do you think it's worth extending the function's doc here to further explain what callers are supposed to pass here? The fee limit in reward addresses is pretty obscure stuff that no one messes around with, so they are hardcoded everywhere as 0x5800. 

Changing the API to receive a sort of "target fee limit" doesn't make it immediately clearer (IMO) and this API has the downside that a caller could invoke it expecting the limit to be one while allowing the actual limit to be much higher (i.e. calling with a `revocationFeeLimit` of `2^23 + 1` actually allows a revocation fee of up to `2^24` atoms).

Reply:
Yeah, I think that makes sense.  I'll get this updated tomorrow.

Reply:
How does this sound?
```
	//
	// Note that fee limits are encoded in the commitment script in terms of the
	// closest base 2 exponent that results in a limit that is >= the provided
	// limit.  In other words, the limits are rounded up to the next power of 2
	// when they are not already an exact power of 2.  For example, a revocation
	// limit of 2^23 + 1 will result in allowing a revocation fee of up to 2^24
	// atoms.
```